pkg/kubelet/logs/container_log_manager.go

```diff
@@ -470,3 +481,82 @@ func (c *containerLogManager) rotateLatestLog(ctx context.Context, id, log strin
 	}
 	return nil
 }
+
+// splitAndRotateLatestLog splits a log file into multiple smaller files if it exceeds MaxSize
+func (c *containerLogManager) splitAndRotateLatestLog(log string) ([]string, error) {
+	// Open the log file for reading
+	r, err := c.osInterface.Open(log)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open log %q: %v", log, err)
+	}
+	defer r.Close()
+
+	var parts []string
+	var partIndex int
+	var currentPart []byte
+	buffer := make([]byte, 1024) // Read in chunks of 1 KB
+
+	for {
+		n, err := r.Read(buffer)
+		if err != nil && err.Error() != "EOF" {
+			return nil, fmt.Errorf("failed to read from log %q: %v", log, err)
+		}
+		if n == 0 {
+			break
+		}
+
+		currentPart = append(currentPart, buffer[:n]...)
+
+		if len(currentPart) > int(c.policy.MaxSize) {
+			partName := fmt.Sprintf("%s.part%d", log, partIndex)
+			parts, err = c.writeSplitPart(partName, parts, currentPart)
```

Comment:
Would it make sense to write parts in a separate goroutines? It would make the whole process a bit faster I guess.